### PR TITLE
[ZEPPELIN-328] Interpreter page should clarify the % magic syntax for interpreter group.name

### PR DESCRIPTION
**Pull request by minahlee**
* Display %interpreterGroup.interpreterName instead of %settingName.interpreterName

  Before
  

  After
  

* Clarify available interpreter as `%group.name` in each notebook when interpreter group name is different from interpreter name.
  * %hql -> %hive.hql
  * %tql -> %tajo.tql
  * %ignite, %ignitesql -> %ignite, %ignite.ignitesql

  Before
  

  After

### CODE REVIEW

**Comment by felixcheung** [comment]
Changes look good
I think in order to match the interpreter group omit rule it should be
```
%hql          -- not work
%hive.hql  -- work, though, this is not necessary
%hive        -- work
```

**Comment by minahlee** [comment]
@felixcheung Thanks for review. I made it to omit interpreter name when it is the first interpreter of the group.

```
%hive.hql -> %hive
%tajo.hql -> %tajo
```

**Comment by corneadoug** [comment]
good

**Comment by felixcheung** [comment]
Is spark-local a group? it seems to have the same group name as the default group (which is also spark)?

File: zeppelin-web/src/app/interpreter/interpreter.html [hunk 1]
```diff
@@ -41,12 +41,13 @@ <h3 class="new_h3" style="float:left">
...
           <small>
-            <span ng-repeat="interpreter in setting.interpreterGroup"
+            <span style="display:inline-block" ng-repeat="interpreter in setting.interpreterGroup"
                   title="{{interpreter.class}}">
-              <span ng-show="$parent.$first && $first">%{{setting.name}} (default)</span>
-              <span ng-show="$parent.$first && !$first">, %{{interpreter.name}}</span>
-              <span ng-show="!$parent.$first && $first">%{{setting.name}}</span>
-              <span ng-show="!$parent.$first && !$first">, %{{setting.name}}.{{interpreter.name}}</span>
+              <span ng-show="!$first">, </span>
+              %<span ng-show="!$parent.$first || $first">{{setting.group}}</span
```
Comment (felixcheung) [inline zeppelin-web/src/app/interpreter/interpreter.html]: this should be `%<span ng-show="$first">`?

File: zeppelin-web/src/app/interpreter/interpreter.html [hunk 2]
```diff
@@ -41,12 +41,13 @@ <h3 class="new_h3" style="float:left">
...
+              ><span ng-show="(!$parent.$first || $first) && !$first">.</span
```
Comment (felixcheung) [inline zeppelin-web/src/app/interpreter/interpreter.html]: this should be `%<span ng-show="!$first">`?

File: zeppelin-web/src/app/interpreter/interpreter.html (hunk 1 above)
Comment (minahlee) [inline zeppelin-web/src/app/interpreter/interpreter.html]: I followed group name omitting rule that is explained in [docs](https://zeppelin.incubator.apache.org/docs/development/writingzeppelininterpreter.html)
`!$parent.$first` is for the rule "You can only omit your interpreter group when your interpreter group is selected as a default group."

**Comment by minahlee** [comment]
@felixcheung AFAIK if two spark interpreters are bind to one notebook, Zeppelin will use only one spark interpreter which is placed upper than the other. And group name is always `spark`. Way to bind the interpreters to notebook should be improved to reduce confusion but I think this can be handled in another PR.
Attaching image to help understanding. In this case Zeppelin use only `spark-local` interpreter and the name of the group is `spark`:

File: zeppelin-web/src/app/interpreter/interpreter.html (hunk 1 above)
Comment (felixcheung) [inline zeppelin-web/src/app/interpreter/interpreter.html]: I understand, I guess I'm not completely sure about the logical expression here.
If we want to show the group name for all group and interpreters that are not the first group, shouldn't we simply check `%<span ng-show="!$parent.$first>` (my previous comment was wrong)?

File: zeppelin-web/src/app/interpreter/interpreter.html (hunk 2 above)
Comment (felixcheung) [inline zeppelin-web/src/app/interpreter/interpreter.html]: Similarly here. To elaborate, if we want to show `.` for all interpreters not the first in the group, then it should be if `!$first`

File: zeppelin-web/src/app/interpreter/interpreter.html (hunk 1 above)
Comment (minahlee) [inline zeppelin-web/src/app/interpreter/interpreter.html]: In my understanding from doc, there are two cases that group name can be omitted.
1. when the interpreter group is selected as a default group.
2. first available interpreter in group.
For example when spark is not set to default group, user doesn't need to write `%spark.spark` but `%spark`.

Comment (felixcheung) [inline zeppelin-web/src/app/interpreter/interpreter.html]: true. ok

**Comment by felixcheung** [comment]
looks good.
agreed multi-binding of the interpreter is confusing. also we might need to support multiple working instances too.

**Comment by Leemoonsoo** [comment]
LGTM. I'm merging it.

**Pull request closed without merging**